From 81de52149719abfe5984fd768bad0801f2aa64f7 Mon Sep 17 00:00:00 2001
From: Shivaprasad Hongal <shongal@codeaurora.org>
Date: Sat, 15 Sep 2018 17:46:23 -0700
Subject: [PATCH 10/11] fs_mgr: Add MF_WRAPPEDKEY flag

Use separate flag to check wrapped key
support for FBE rather than file contents
mode.

CRs-Fixed: 2316112
Change-Id: Ied59d43a82b6fe9157fd2db405de28549e98982c
---
 fs_mgr/fs_mgr_fstab.cpp            | 8 ++++++--
 fs_mgr/fs_mgr_priv.h               | 1 +
 fs_mgr/include_fstab/fstab/fstab.h | 1 +
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/fs_mgr/fs_mgr_fstab.cpp b/fs_mgr/fs_mgr_fstab.cpp
index 23e32b402..7ccec8c8d 100644
--- a/fs_mgr/fs_mgr_fstab.cpp
+++ b/fs_mgr/fs_mgr_fstab.cpp
@@ -83,6 +83,7 @@ static struct flag_list fs_mgr_flags[] = {
     {"encryptable=", MF_CRYPT},
     {"forceencrypt=", MF_FORCECRYPT},
     {"fileencryption=", MF_FILEENCRYPTION},
+    {"wrappedkey", MF_WRAPPEDKEY},
     {"forcefdeorfbe=", MF_FORCEFDEORFBE},
     {"keydirectory=", MF_KEYDIRECTORY},
     {"nonremovable", MF_NONREMOVABLE},
@@ -114,13 +115,11 @@ static struct flag_list fs_mgr_flags[] = {
 #define EM_ICE          2
 #define EM_AES_256_CTS  3
 #define EM_AES_256_HEH  4
-#define EM_ICE_WRAPPED_KEY_SUPPORTED 7
 
 static const struct flag_list file_contents_encryption_modes[] = {
     {"aes-256-xts", EM_AES_256_XTS},
     {"software", EM_AES_256_XTS}, /* alias for backwards compatibility */
     {"ice", EM_ICE}, /* hardware-specific inline cryptographic engine */
-    {"ice_wrapped_key_supported", EM_ICE_WRAPPED_KEY_SUPPORTED}, /* ICE engine with wrapped key support */
     {0, 0},
 };
 
@@ -1022,3 +1021,8 @@ int fs_mgr_has_sysfs_path(const struct fstab_rec *fstab)
 {
     return fstab->fs_mgr_flags & MF_SYSFS;
 }
+
+int fs_mgr_is_wrapped_key_supported(const struct fstab_rec *fstab)
+{
+    return fstab->fs_mgr_flags & MF_WRAPPEDKEY;
+}
diff --git a/fs_mgr/fs_mgr_priv.h b/fs_mgr/fs_mgr_priv.h
index 9011bb38e..c1cc3beb0 100644
--- a/fs_mgr/fs_mgr_priv.h
+++ b/fs_mgr/fs_mgr_priv.h
@@ -111,6 +111,7 @@
 #define MF_AVB             0X2000000
 #define MF_KEYDIRECTORY    0X4000000
 #define MF_SYSFS           0X8000000
+#define MF_WRAPPEDKEY      0X10000000
 
 #define DM_BUF_SIZE 4096
 
diff --git a/fs_mgr/include_fstab/fstab/fstab.h b/fs_mgr/include_fstab/fstab/fstab.h
index 4fd850927..436a233d6 100644
--- a/fs_mgr/include_fstab/fstab/fstab.h
+++ b/fs_mgr/include_fstab/fstab/fstab.h
@@ -87,6 +87,7 @@ int fs_mgr_is_nofail(const struct fstab_rec* fstab);
 int fs_mgr_is_latemount(const struct fstab_rec* fstab);
 int fs_mgr_is_quota(const struct fstab_rec* fstab);
 int fs_mgr_has_sysfs_path(const struct fstab_rec* fstab);
+int fs_mgr_is_wrapped_key_supported(const struct fstab_rec *fstab);
 
 std::string fs_mgr_get_slot_suffix();
 std::set<std::string> fs_mgr_get_boot_devices();
-- 
2.17.1

